src/Tests/Tests/Test_Vibrato.cpp

```diff
@@ -16,6 +15,314 @@ namespace vibrato_test {
+        {
+            tmpOutput[channel] = new float[iBufferLength];
+            CSynthesis::generateNoise(ppfInputBuffer[channel], iBufferLength);
+        }
+
+        EXPECT_EQ(vibrato->process(ppfInputBuffer, ppfOutputBuffer, iBufferLength),Error_t::kNoError);
+        EXPECT_EQ(vibrato->process(ppfInputBuffer,tmpOutput,1021),Error_t::kNoError);
+        for (int c = 0; c < iNumChannels; c++)
+        {
+            CHECK_ARRAY_CLOSE(tmpOutput[c], ppfOutputBuffer[c], iNumChannels, 1E-5);
+        }
+        for(int i=0;i<iNumChannels;i++)
+            delete[] tmpOutput[i];
+        delete[] tmpOutput;
+    }
+
+    TEST_F(VibratoTest, OutputEqualsDelayedInput)
+    {
+        for (int channel = 0; channel < iNumChannels; channel++)
+        {
+            CSynthesis::generateSine(ppfInputBuffer[channel],fFrequencyInHz, fSampleRateInHz, iBufferLength, 1);
+        }
+        vibrato->setParam(CVibrato::CVibratoParam::kWidth, 0);
+        vibrato->process(ppfInputBuffer, ppfOutputBuffer, iBufferLength);
+        int total_delay_samples = static_cast<int>(2 + (fWidthInSec * 2 * fSampleRateInHz));//+ (fDelayInSec * fSampleRate);
+
+
+        for (int channel = 0; channel < iNumChannels; channel++)
+        {
+            CHECK_ARRAY_CLOSE(ppfInputBuffer[channel] + total_delay_samples, ppfOutputBuffer[channel],
+                              iBufferLength - total_delay_samples, 1e-3);
+        }
+    }
+
+    TEST_F(VibratoTest, DCStaysDC)
+    {
+        for (int channel = 0; channel < iNumChannels; channel++)
+        {
+            CSynthesis::generateDc(ppfInputBuffer[channel], iBufferLength, 1);
+        }
+        vibrato->process(ppfInputBuffer, ppfOutputBuffer, iBufferLength);
+        // Since the output file will be delayed, we need to calculate how many samples it is delayed by and offset that when checking the buffers
+        // L=2+DELAY+WIDTH*2; - delay accounted for elsewhere? Trial and error showed the below value is right
+        int total_delay_samples = static_cast<int>(2 + (fWidthInSec * 2 * fSampleRateInHz));//+ (fDelayInSec * fSampleRate);
+
+
+        for (int channel = 0; channel < iNumChannels; channel++)
+        {
+            CHECK_ARRAY_CLOSE(ppfInputBuffer[channel] + total_delay_samples, ppfOutputBuffer[channel],
+                              iBufferLength - total_delay_samples, 1e-3);
+        }
+    }
+
+    TEST_F(VibratoTest, ZeroInputTest)
+    {
+        float** tmpOutput = new float* [iNumChannels];
+        for (int channel = 0; channel < iNumChannels; channel++)
+        {
+            tmpOutput[channel] = new float[iBufferLength];
+            for (int i = 0; i < iBufferLength; i++)
+            {
+                tmpOutput[channel][i] = 0;
+            }
+
+        }
+
+        EXPECT_EQ(vibrato->process(ppfInputBuffer, tmpOutput, 1024), Error_t::kNoError);
+
+        for (int i = 0; i < iNumChannels; i++)
+            delete[] tmpOutput[i];
+        delete[] tmpOutput;
+    }
```

Comment:
Maybe one additional test can be that checking if any glitch shows up after vibrato. Normally there won't be this scenario if fractional buffer is working correctly. It is more like a double check on the output audio.